numpy/__init__.pyi

```diff
@@ -2986,6 +3035,8 @@ int0 = signedinteger[_NBitIntP]
 int_ = signedinteger[_NBitInt]
 longlong = signedinteger[_NBitLongLong]
 
+# TODO: `item`/`tolist` returns either `dt.timedelta` or `int`
+# depending on the unit
```

Comment:
There are a couple of cases where the return-type is dependant on the unit (`datetime64` and `timedelta64`) or some other underlying object (`void`and `object_`). 
In the long term, it might be worthwhile to make these classes [generic](https://docs.python.org/dev/glossary.html#term-generic-type) w.r.t. a suitable parameter (_e.g._ the unit) to sort this out.